src/core/probe/bpf/include/common_defs.h

```diff
@@ -32,6 +31,52 @@ static __always_inline bool collection_enabled() {
 #define COMMON_SECTION_CORE	0
 #define COMMON_SECTION_TASK	1
 
+/* Current log level. */
+const volatile u8 log_level = 0;
+
+/* Aligned with the log crate. */
+enum {
+	LOG_ERROR = 1,
+	LOG_WARN,
+	LOG_INFO,
+	LOG_DEBUG,
+	LOG_TRACE,
+};
+
+/* Checks if the level will be reported. */
+#define retis_log_enabled(__lvl)			\
+({							\
+	bool active = false;				\
+	if (log_level && __lvl <= log_level)		\
```

Comment:
Do you think allowing no log at all has benefits? Otherwise we can simplify the above (removing it and open coding it below) and defaults `log_level` to `LOG_ERROR`.

Reply:
`log_level` should never be 0 unless we explicitly set it so with `set_max_level()`.
It always defaults to `log::max_level()` and so the same level of user space.
In general, I think it's nice to have such a direct mapping.

Maybe we want to rename `retis_log_enabled()` to `retis_log_allowed()` (I'm open to proposals :) as the current name doesn't perfectly describe well the macro?

Reply:
> `log_level` should never be 0 unless we explicitly set it so with `set_max_level()`. It always defaults to `log::max_level()` and so the same level of user space. In general, I think it's nice to have such a direct mapping.

Right, the point was then we can remove the `if (log_level && ` part from above as it's always true. And then we can open code it for better readability. In addition, to make the default clear we can initialize `log_level` to `LOG_ERROR`.

> Maybe we want to rename `retis_log_enabled()` to `retis_log_allowed()` (I'm open to proposals :) as the current name doesn't perfectly describe well the macro?

I don't have a strong feeling on this one :)

Reply:
> > `log_level` should never be 0 unless we explicitly set it so with `set_max_level()`. It always defaults to `log::max_level()` and so the same level of user space. In general, I think it's nice to have such a direct mapping.
> 

makes sense to remove `log_level &&` as it's not really useful.

> Right, the point was then we can remove the `if (log_level && ` part from above as it's always true. And then we can open code `retis_log_enabled` for better readability. In addition, to make the default clear we can initialize `log_level` to `LOG_ERROR`.
> 
> > Maybe we want to rename `retis_log_enabled()` to `retis_log_allowed()` (I'm open to proposals :) as the current name doesn't perfectly describe well the macro?
> 
> I don't have a strong opinion on this one :)

Let's open code it and not think about the name  :)

Reply:
about `log_level` to `LOG_ERROR`, user space defaults to LOG_INFO. That would be misaligned.

Reply:
Right, we should make the two default to the same value at least here for the initial submission.